File: src/Vult.WebApp/projects/vult/src/app/core/models/testimonial.ts
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

export interface Testimonial {
  testimonialId: string;
  customerName: string;
  photoUrl: string;
  rating: number;
  text: string;
  createdDate: string;
  updatedDate: string;
}

export interface CreateTestimonialRequest {
  customerName: string;
  photoUrl: string;
  rating: number;
  text: string;
}

export interface UpdateTestimonialRequest {
  testimonialId: string;
  customerName: string;
  photoUrl: string;
  rating: number;
  text: string;
}

export interface TestimonialResponse {
  testimonial: Testimonial;
  success: boolean;
  errors: string[];
}

export interface GetTestimonialByIdResponse {
  testimonial: Testimonial;
}

File: src/Services/TestimonialService/TestimonialService.Api/Model/Testimonial.cs
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See LICENSE in the project root for license information.

namespace TestimonialService.Api.Model;

public class Testimonial
{
    public Guid TestimonialId { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public string PhotoUrl { get; set; } = string.Empty;
    public int Rating { get; set; }
    public string Text { get; set; } = string.Empty;
    public DateTime CreatedDate { get; set; }
    public DateTime UpdatedDate { get; set; }
}
